worker/draft.go

```diff
@@ -1837,6 +1853,11 @@ func (n *node) InitAndStartNode() {
 			n.SetRaft(raft.StartNode(n.Cfg, peers))
 			// Trigger election, so this node can become the leader of this single-node cluster.
 			n.canCampaign = true
+			// Also trigger a snapshot so that this node can take a snapshot if required
+			// Need a delay otherwise it interfers with starting of Raft loop
+			time.AfterFunc(1*time.Second, func() {
```

Comment:
Can i ask how it might interfere with the starting of Raft loop?

Reply:
yeah, and we should avoid it. This will turn into a bad idea, soon enough.

Reply:
When the node initializes, it begins in a follower state, even in a single alpha cluster scenario. This pause ensures that the leader election process occurs, allowing the node to acquire leadership. Without this pause, the node tries to capture a snapshot but fails the check in [n.AmLeader()](https://github.com/dgraph-io/dgraph/blob/13e2f77075407437f0a7cb06df81f866644d28d2/worker/draft.go#L1062), resulting in no snapshot being taken.